src/middleware/otpauthMiddleware.ts

```diff
@@ -0,0 +1,64 @@
+import randomstring from 'randomstring';
+import Jwt, { VerifyErrors } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import { config } from 'dotenv';
+import { sendEmail } from '../utils/email';
+import { DataInfo } from '../controllers/OTPcontroller';
+import { decodeToken, verifyToken } from '../utils/tokenGenerate';
+
+config();
+
+export const generateOtpToken = async (id: string, email: string) => {
+  const code = await randomstring.generate({
+    length: 6,
+    charset: 'numeric',
+  });
+  
+  const body = {
+    id,
+    email,
+    otp: code,
+  };
+  const secret = process.env.JWT_SECRET;
+  const token = await Jwt.sign({ body }, secret as string, {
+    expiresIn: '1h',
+  });
+  const mailOptions = {
+    to: email,
+    subject: 'OTP Email verification',
+    template: 'otp',
+    context: {
+      otp: `${code}`,
+      verifyLInk: `http://localhost:3000/api/users/verify/${token}`
```

Comment:
please use Link which is dynamic from env

Reply:
okay